### PR TITLE
storage: move to pkg/kv/kvserver

### PR DESCRIPTION
**Pull request by tbg**
```
pkg/kv/kvserver <-- today: pkg/storage
pkg/storage     <-- today: pkg/storage/engine
```

### CODE REVIEW

**Comment by tbg** [comment]
Heh, I botched the rename (I used kvserver in one commit, storage in the other) but I just wanted to see if CI can become green anyway.

File: pkg/testutils/lint/lint_test.go [hunk 1]
```diff
@@ -1018,7 +1018,7 @@ func TestLint(t *testing.T) {
 		if err := stream.ForEach(stream.Sequence(
 			filter,
 			stream.GrepNot(`.*\.lock`),
-			stream.GrepNot(`^storage\/engine\/rocksdb_error_dict\.go$`),
+			stream.GrepNot(`^storage\/rocksdb_error_dict\.go$`),
```
Comment (nvanbenschoten) [inline pkg/testutils/lint/lint_test.go]: Not for now, but what's up with this? Don't we have the same grep twice now?

**Review by nvb — APPROVED** [review]
But I know you have more coming down the pipeline.

**Comment by tbg** [comment]
bors r=nvanbenschoten

I ran a number of TC invocations. The last known problem was that the mixed version tests were failing, which is now fixed (verified locally). Going to try this.

**Comment by tbg** [comment]
Both CI runs failed on TestSecondary, which is flaky on master and fixed in http://github.com/cockroachdb/cockroach/pull/45695. That PR is now merging, so retrying.

I also added a commit that skips testrace if more than 10 packages changed. I will revisit that when this PR merges, too, but it is necessary to avoid blocking bors for >2h.

bors r=nvanbenschoten

**Comment by tbg** [comment]
I caught a merge conflict with f13e05d956d8375b3f5ddb6854539195cc3ca700 (which is bors'ing right now and which I need to merge to get the flake out of the way).

I'm changing this PR to revert this commit first, then cherry-pick it again (including resolution of the merge conflict).

**Comment by tbg** [comment]
bors r=nvanbenschoten

**Comment by tbg** [comment]
bors r+

Messed something up in the "skip race" commit. Aborting this early, so much stuff will be queued in 2h

**Comment by tbg** [comment]
The bors CI ran through (https://teamcity.cockroachdb.com/viewLog.html?buildId=1785201&buildTypeId=Cockroach_UnitTests&tab=buildResultsDiv&branch_Cockroach=staging) but then threw up because the license/cla webhook hadn't fired.
I'm calling this fair game for a manual merge.

**Pull request merged**